dev/breeze/src/airflow_breeze/utils/run_utils.py

```diff
@@ -40,12 +41,22 @@ def run_command(
     verbose: bool = False,
     suppress_raise_exception: bool = False,
     suppress_console_print: bool = False,
+    env: Optional[Mapping[str, str]] = None,
```

Comment:
This is the "gist" and the problem you had been struggling with. Instead of modifying "os.environ" I pass the variables as dictionary to the "run_command".